### PR TITLE
3640 jsonqueue unit tests

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|#3640|

## Description

These are the unit tests for the new functions implemented in the PR #4256.

### CODE REVIEW

File: src/unit_tests/test_json_queue.c [hunk 1]
```diff
@@ -0,0 +1,1070 @@
...
+{
+    (void) state;
+
+    file_queue *fileq;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+
+    /* fopen fail */
+
+    jqueue_init(fileq);
```
Comment (Molter73) [inline src/unit_tests/test_json_queue.c]: I would move the os_calloc and jqueue_init into a setup function for this test and pass it via `state`.
The ending free statement should also be moved into a teardown function, by doing this we guarantee that the alloc'd memory will be freed regardless of the test passing or failing.
Also, replace calls to wazuh specific code (such as os_calloc) with std function calls.

File: src/unit_tests/test_json_queue.c [hunk 2]
```diff
@@ -0,0 +1,1070 @@
...
+
+void test_jqueue_open_fail_fseek(void **state)
+{
+    (void) state;
+
+    file_queue *fileq;
+    FILE *fp;
+
+    os_calloc(1, sizeof(file_queue), fileq);
+    os_calloc(1, sizeof(FILE), fp);
```
Comment (Molter73) [inline src/unit_tests/test_json_queue.c]: Same as with test_jqueue_open_fail_fopen, if several variables need to be alloc'd they can be grouped into a structure and passed via `state`.

File: src/unit_tests/test_json_queue.c (hunk 1 above)
Comment (TomasTurina) [reply src/unit_tests/test_json_queue.c]: Done!

File: src/unit_tests/test_json_queue.c (hunk 2 above)
Comment (TomasTurina) [reply src/unit_tests/test_json_queue.c]: Done!

File: src/unit_tests/test_json_queue.c [hunk 3]
```diff
@@ -0,0 +1,931 @@
...
+    will_return(__wrap_fgets, "ok");
+
+    aux->json = jqueue_next(fileq);
+
+    assert_string_equal(cJSON_GetObjectItem(aux->json, "Test")->valuestring, "Hello World 2");
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_ptr_equal(fileq->fp, fp);
+}
+
+void test_jqueue_close(void **state)
```
Comment (Molter73) [inline src/unit_tests/test_json_queue.c]: Calling jqueue_open in this test is not necessary, we can simply pass a mock fileq with a random fp value and check it is NULL on return.

File: src/unit_tests/test_json_queue.c [hunk 4]
```diff
@@ -0,0 +1,931 @@
...
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, fp);
+    will_return(__wrap_fstat, 1);
+
+    int ret = Handle_JQueue(fileq, CRALERT_READ_ALL);
+
+    assert_int_equal(ret, 1);
+    assert_ptr_equal(fileq->fp, fp);
+}
+
```
Comment (Molter73) [inline src/unit_tests/test_json_queue.c]: A test having both CRALERT_READ_ALL and CRALERT_FP_SET could be written.

**Pull request merged**